src/components/dialog/dialog.js

```diff
@@ -393,6 +393,14 @@ function MdDialogProvider($$interimElementProvider) {
       }
     };
 
+    function trapFocus(ev) {
+      var dialog = document.querySelector('md-dialog');
+
+      if (dialog && !dialog.contains(ev.target)) {
+        ev.stopPropagation();
```

Comment:
I didn't think it needed to stop other listeners on the same element, just prevent the listener triggering this function from bubbling/capturing.